### PR TITLE
gpui: Fix SVG color render, when color have alpha

### PR DESCRIPTION
**Pull request by huacnlee**
Release Notes:

- N/A

## Demo

- [Source SVG](https://github.com/user-attachments/assets/1c681e01-baba-4613-a3e7-ea5cb3015406) click here open in browser.

| Before | After |
| --- | --- |
| image | image |

### CODE REVIEW

File: crates/gpui/Cargo.toml [hunk 1]
```diff
@@ -116,6 +116,7 @@ env_logger.workspace = true
 rand.workspace = true
 util = { workspace = true, features = ["test-support"] }
 http_client = { workspace = true, features = ["test-support"] }
+reqwest_client = { workspace = true }
```
Comment (huacnlee) [inline crates/gpui/Cargo.toml:119]: I have to add this to gpui example, to let remote image works.

File: crates/gpui/src/platform/mac/text_system.rs [hunk 2]
```diff
@@ -418,11 +419,7 @@ impl MacTextSystemState {
             if params.is_emoji {
                 // Convert from RGBA with premultiplied alpha to BGRA with straight alpha.
                 for pixel in bytes.chunks_exact_mut(4) {
-                    pixel.swap(0, 2);
-                    let a = pixel[3] as f32 / 255.;
-                    pixel[0] = (pixel[0] as f32 / a) as u8;
-                    pixel[1] = (pixel[1] as f32 / a) as u8;
-                    pixel[2] = (pixel[2] as f32 / a) as u8;
+                    swap_rgba_pa_to_bgra(pixel);
```
Comment (huacnlee) [inline crates/gpui/src/platform/mac/text_system.rs:422]: Extract this code into a method to color.rs and fix divide 0 issue.

File: crates/gpui/Cargo.toml (hunk 1 above)
Comment (SomeoneToIgnore) [reply crates/gpui/Cargo.toml:119]: This does not look good at all, as it's a general library that got some odd, fat client library inside.
Which also uses `tokio` underneath, that is not at all used in Zed which uses gpui.
I understand this is a dev dependency, but Zed uses gpui in tests too, so nope.

We can add a regular `cargo init` thing into `examples` dir and some readme mention that it's a special directory that you run manually, and specify whatever odd dependencies there.
Or hide it behind a feature flag at the very least.

Comment (huacnlee) [reply crates/gpui/Cargo.toml:119]: I am going to remove this, this PR not requirement of this.

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you!

**Pull request merged**